service/history/task/timer_standby_task_executor.go

```diff
@@ -110,7 +110,10 @@ func (t *timerStandbyTaskExecutor) Execute(
 	case persistence.TaskTypeWorkflowBackoffTimer:
 		return t.executeWorkflowBackoffTimerTask(ctx, timerTask)
 	case persistence.TaskTypeDeleteHistoryEvent:
-		return t.executeDeleteHistoryEventTask(ctx, timerTask)
+		// special timeout for delete history event
+		deleteHistoryEventContext, deleteHistoryEventCancel := context.WithTimeout(context.Background(), taskDeleteHistoryEventContextTimeout)
```

Comment:
I know the existing `ctx` has 3 sec timeout and doesn't listen for queue shut down signals. 
With the new 60s timeout for history deletions it is going to be more likely to end up hanging a shutdown. We should consider making these operations responsive to shut down signals.
@Shaddoll what do you think?

Reply:
Discussed offline. We will hook up Stop signals to task executor in a follow up PR. 
In this PR, let's make this 60s dynamically configurable